oshi-core/src/main/java/oshi/software/os/unix/solaris/SolarisOperatingSystem.java

```diff
@@ -129,32 +130,49 @@ public OSProcess getProcess(int pid) {
 
     @Override
     public List<OSProcess> queryDescendantProcesses(int parentPid) {
-        Set<String> descendantPids = new HashSet<>();
-        addChildrenToDescendantSet(Integer.toString(parentPid), descendantPids, true);
+        Set<String> descendantPids = getChildrenOrDescendants(Integer.toString(parentPid), true);
         if (descendantPids.isEmpty()) {
             return Collections.emptyList();
         }
         return getProcessListFromPS("ps -o s,pid,ppid,user,uid,group,gid,nlwp,pri,vsz,rss,etime,time,comm,args -p "
                 + String.join(",", descendantPids), -1);
     }
 
-    private static void addChildrenToDescendantSet(String parentPid, Set<String> descendantPids, boolean recurse) {
-        // Get list of children
+    private static Set<String> getChildren(String parentPid) {
         Set<String> childPids = new HashSet<>();
         for (String s : ExecutingCommand.runNative("pgrep -P " + parentPid)) {
```

Comment:
see earlier comments about only using this for the one-generation branch and using the superclass method for all descendants.